src/main/java/com/vint/shop/service/impl/UserDetailsServiceImpl.java

```diff
@@ -10,42 +12,70 @@
 import org.springframework.security.core.userdetails.UserDetails;
 import org.springframework.security.core.userdetails.UserDetailsService;
 import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
-import java.util.HashSet;
-import java.util.Objects;
-import java.util.Set;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import java.util.*;
+
 
 @Service
 public class UserDetailsServiceImpl implements UserDetailsService {
```

Comment:
In general, it's better to differ UserDetailsService and UserService which is responsible for User bean

Reply:
Ok, what is the best name for it?